src/reactive-membrane.ts

```diff
@@ -110,36 +107,18 @@ export class ReactiveMembrane {
         return unwrap(p);
     }
 
-    private getReactiveState(value: any, distortedValue: any): ReactiveState {
-        const {
-            objectGraph,
-        } = this;
-        let reactiveState = objectGraph.get(distortedValue);
-        if (reactiveState) {
-            return reactiveState;
+    private getReactiveHandler(value: any, distortedValue: any, readOnly: boolean): any {
```

Comment:
avoid boolean arguments, instead just pass the corresponding map, that should also make the code more readable.

Reply:
or make it two methods

Reply:
I think I rather prefer two methods.

Reply:
I can make it two methods. It just requires duplicating ~5 lines of code, which is no big deal.